java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java

```diff
@@ -31,18 +41,61 @@ public MenuControllerImpl(MenuService menuService) {
                                                        final Integer pageSize,
                                                        final Integer pageNumber) {
 
+        List<SearchMenuResultItem> menuResultItems;
+        if (StringUtils.isNotEmpty(searchTerm) && Objects.nonNull(restaurantId) ) {
+            return ResponseEntity.ok(
+                    new SearchMenuResult(pageSize, pageNumber, getResultsByRestaurantIdAndNameContaining(searchTerm,restaurantId, pageSize, pageNumber)));
+        }
+
+        if (StringUtils.isNotEmpty(searchTerm)) {
+            return ResponseEntity.ok(
+                    new SearchMenuResult(pageSize, pageNumber, getResultsByNameContaining(searchTerm, pageSize, pageNumber)));
+        }
+
+        if (Objects.nonNull(restaurantId)) {
+            return ResponseEntity.ok(
+                    new SearchMenuResult(pageSize, pageNumber, getAllResultsByRestaurantId(restaurantId, pageSize, pageNumber)));
+        }
+
         List<Menu> menus = menuService.all(pageNumber, pageSize);
-        List<SearchMenuResultItem> menuResultItems = menus.stream()
+        menuResultItems = menus.stream()
                 .map(SearchMenuResultItem::new)
                 .collect(Collectors.toList());
 
         return ResponseEntity.ok(
                 new SearchMenuResult(pageSize, pageNumber, menuResultItems));
     }
 
+    private List<SearchMenuResultItem> getAllResultsByRestaurantId(UUID restaurantId, Integer pageSize, Integer pageNumber) {
+        final Sort sort = Sort.by(Sort.Direction.ASC, "restaurantId");
+        final CosmosPageRequest pageRequest = new CosmosPageRequest(pageNumber, pageSize, null, sort);
```

Comment:
This should all be in the service. 
Cosmos dependencies should definitely not be at this level.